packages/app/src/cli/services/environment.ts

```diff
@@ -69,6 +59,58 @@ interface DevEnvironmentOutput {
   tunnelPlugin: string | undefined
 }
 
+/**
+ * Make sure there is a valid environment to execute `generate extension`
+ *
+ * We just need a valid app API key to access the Specifications API.
+ * - If the API key is provided via flag, we use it.
+ * - Else, if there is cached API key for the current directory, we use it.
+ * - Else, we prompt the user to select/create an app.
+ *
+ * The selection is then cached as the "dev" app for the current directory.
+ */
+export async function ensureGenerateEnvironment(options: {
+  apiKey?: string
+  directory: string
+  reset: boolean
+  token: string
+}): Promise<string> {
+  if (options.apiKey) {
+    const app = await fetchAppFromApiKey(options.apiKey, options.token)
+    if (!app) throw InvalidApiKeyError(options.apiKey)
+    return app.apiKey
+  }
+  const cachedInfo = await getAppDevCachedInfo({reset: options.reset, directory: options.directory})
+
+  if (cachedInfo === undefined && !options.reset) {
+    const explanation =
+      `\nLooks like this is the first time you're running 'generate extension' for this project.\n` +
+      'Configure your preferences by answering a few questions.\n'
+    output.info(explanation)
+  }
+
+  if (cachedInfo?.appId && cachedInfo?.orgId) {
+    const org = await fetchOrgFromId(cachedInfo.orgId, options.token)
+    const app = await fetchAppFromApiKey(cachedInfo.appId, options.token)
+    if (!app || !org) throw InvalidApiKeyError(cachedInfo.appId)
+    const packageManager = await getPackageManager(options.directory)
+    showGenerateReusedValues(org.businessName, cachedInfo, packageManager)
+    return app.apiKey
+  } else {
+    const orgId = cachedInfo?.orgId || (await selectOrg(options.token))
+    const {organization, apps} = await fetchOrgAndApps(orgId, options.token)
+    const localAppName = await loadAppName(options.directory)
+    const selectedApp = await selectOrCreateApp(localAppName, apps, organization, options.token, cachedInfo?.appId)
+    await store.setAppInfo({
+      appId: selectedApp.apiKey,
+      title: selectedApp.title,
+      directory: options.directory,
+      orgId,
+    })
+    return selectedApp.apiKey
+  }
+}
```

Comment:
This workflow seems very similar to the one that we follow for `dev`. Have you considered reusing it to prevent potential inconsistencies down the road?